src/primitives/compression.cpp

```diff
@@ -0,0 +1,358 @@
+#include <primitives/compression.h>
+
+CCompressedTxId::CCompressedTxId(const uint256& txid) : m_block_height(0), m_block_index(0), m_txid(txid) {};
+
+CCompressedTxId::CCompressedTxId(const uint32_t& block_height, const uint32_t& block_index) : m_block_height(block_height), m_block_index(block_index) {};
+
+std::string CCompressedTxId::ToString() const
+{
+    return strprintf("CCompressedTxId(block_height=%u, block_index=%u, txid=%s)",
+    m_block_height,
+    m_block_index,
+    HexStr(m_txid));
```

Comment:
In 89b52782f8d6aca75d49048da1e137fbf615c0e6:

nit: weird formatting here and on other printf calls

Reply:
I matched it to the printf calls in transaction.cpp and can't see any divergences.

Reply:
I guess you mean `strprintf`, and I don't -see any in transaction.cpp that are spread across multiple lines with no further indentation like this.

Reply:
I fixed the CompressedTransaction ToString to include indentation but I can't properly nest indentation for the nested classes such as the prevout and Txid.